src/auto_optional/code_processing.py

```diff
@@ -99,27 +99,56 @@ def leave_Module(
 
         Is only used to add the ``typing.Optional`` import statement when missing.
         """
-        if self.optional_import_checker.import_names and self.optional_import_needed:
-            import_statement = cst.SimpleStatementLine(
-                body=[
-                    cst.ImportFrom(
-                        module=cst.Name(
-                            value="typing",
-                        ),
-                        names=[
-                            cst.ImportAlias(
-                                name=cst.Name(
-                                    value="Optional",
-                                ),
+        if (
+            not self.optional_import_checker.import_names
+            or not self.optional_import_needed
+        ):
+            return updated_node
+
+        import_statement = cst.SimpleStatementLine(
+            body=[
+                cst.ImportFrom(
+                    module=cst.Name(
+                        value="typing",
+                    ),
+                    names=[
+                        cst.ImportAlias(
+                            name=cst.Name(
+                                value="Optional",
                             ),
-                        ],
+                        ),
+                    ],
+                )
+            ]
+        )
+        insert_at = 0
+        for index, body_part in enumerate(updated_node.body):
+            if (
+                isinstance(body_part, cst.SimpleStatementLine)
+                and len(body_part.children) > 0
+                and (
+                    (  # module level docstring
+                        index == 0
+                        and m.matches(body_part.children[0], m.Expr(m.SimpleString()))
                     )
-                ]
-            )
-            return updated_node.with_changes(
-                body=[import_statement, *updated_node.body]
-            )
-        return updated_node
+                    or m.matches(  # future import
+                        body_part.children[0],
+                        m.ImportFrom(module=m.Name("__future__")),
+                    )
+                )
+            ):
+                # module docstring
+                insert_at += 1
+                continue
+            if index > 1:
+                break
```

Comment:
Could you extract this into a helper method? something descriptive like `__find_body_index_for_option_import_insertion`?

Reply:
I can do that tomorrow. I've allowed edits of this PR by maintainers, if you'd rather have it fixed today.